Week3/database.js

```diff
@@ -0,0 +1,106 @@
+const mysql = require('mysql');
+const util = require('util');
+const connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  database: 'todo_data'
+});
+
+const execQuery = util.promisify(connection.query.bind(connection));
+
+const getAll = async function (id) {
+  let sql = `select * from users a join toDosList b on a.Id = b.UserId join todoItems c on b.listId = c.listId where a.Id = ?`
+  try {
+    return await execQuery(sql, [id]);
+  } catch (error) {
+    connection.end();
+    return error;
+  }
+};
+const createUser = async function (id, firstName, lastName) {
```

Comment:
With the auto-increment, you won't need to pass an id here. In real-world cases, the client, ex.: web page or mobile app, would not know the id for a new user as this will be the responsibility of the server. In this case, our MySQL server will control that for us.